src/python/pants/core/util_rules/external_tool.py

```diff
@@ -152,6 +175,7 @@ def generate_exe(self, plat: Platform) -> str:
 
     def get_request(self, plat: Platform) -> ExternalToolRequest:
         """Generate a request for this tool."""
+        self.check_version_constraint(self.version)
```

Comment:
I first implemented the check as a member type of the known-versions option, and had it working (especially well with my other PR regarding checking the validity of option member values) but ran aground on the limitation for the types accepted for `member_type`. Also, I think this will be more robust, with all validation taking place with the actual value. Only downside is that we don't get to run the check until at the last moment, where as if the check was on known versions, you'd get the indication as soon as you use it in your configuration..

Reply:
Commented above: I think that you can move the check into the constructor for the `Subsystem`.